src/core/server/http/http_config.ts

```diff
@@ -70,6 +70,18 @@ const configSchema = schema.object(
     securityResponseHeaders: securityResponseHeadersSchema,
     customResponseHeaders: schema.recordOf(schema.string(), schema.any(), {
       defaultValue: {},
+      validate(value) {
+        const RESPONSE_HEADER_DENY_LIST = ['location'];
```

Comment:
I debated placing this somewhere else, but this is the only place the list is used. The extra object allocation each time `validate` is called should not have any measurable impact.